src/test/java/com/microsoft/azure/cosmosdb/kafka/connect/sink/integration/SinkConnectorIT.java

```diff
@@ -0,0 +1,263 @@
+package com.microsoft.azure.cosmosdb.kafka.connect.sink.integration;
+
+import com.azure.cosmos.CosmosClient;
+import com.azure.cosmos.CosmosClientBuilder;
+import com.azure.cosmos.CosmosContainer;
+import com.azure.cosmos.CosmosDatabase;
+import com.azure.cosmos.models.CosmosContainerProperties;
+import com.azure.cosmos.models.CosmosQueryRequestOptions;
+import com.azure.cosmos.models.ThroughputProperties;
+import com.azure.cosmos.util.CosmosPagedIterable;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.lang3.RandomUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.apache.kafka.connect.json.JsonSerializer;
+import org.sourcelab.kafka.connect.apiclient.Configuration;
+import org.sourcelab.kafka.connect.apiclient.KafkaConnectClient;
+import org.sourcelab.kafka.connect.apiclient.request.dto.NewConnectorDefinition;
+import com.microsoft.azure.cosmosdb.kafka.connect.IntegrationTest;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.concurrent.ExecutionException;
+
+import static org.apache.kafka.common.utils.Utils.sleep;
+import static org.sourcelab.kafka.connect.apiclient.request.dto.NewConnectorDefinition.Builder;
+
+/**
+ * Integration tests for the CosmosDB sink connector.
+ * Tests that messages posted to Kafka propogate to CosmosDB.
+ * 
+ * NOTE: Requires a Cosmos DB instance and the docker-compose orchestration to be running.
```

Comment:
have we tested this using the Cosmos DB emulator? it should work, but it would be good to test that during an integration test, we can spin up an ACI (or Docker image) with the emulator installed and test against the emulator. this means that our integration tests don't require an actual Cosmos DB instance to be running anywhere. 

this might be something we'd have to check if we can do in github actions when the integration tests run.

Reply:
I was looking into this before for the devcontainer setup and found out that the cosmosdb emulator will only work on windows machines, whether on docker or natively. for a mac/linux machine, this would require a windows vm and that increases the complexity https://docs.microsoft.com/en-us/azure/cosmos-db/local-emulator?tabs=cli%2Cssl-netstd21#run-on-linux-macos

if github actions can support a windows runtime environment, then it's certainly possible to run emulator and the tests against it. I'll look into this.

Reply:
true. not a big deal. i wouldn't burn any cycles on this for now. apparently the emulator for Linux is a work in progress.